### PR TITLE
Disable space-before-function-paren

### PR DESCRIPTION
**Pull request by cmatheson**
This gets prettier and eslint on the same page.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -30,6 +30,7 @@ module.exports = {
     }],
     'react-hooks/rules-of-hooks': 'error',
     'react-hooks/exhaustive-deps': 'warn',
+    'space-before-function-paren': 'off',
```
Comment (cmatheson) [inline index.js]: this could also be 'never' but i figured we could just let prettier take care of it. let me know if i should change it.

Comment (tpennock) [reply index.js]: I've always preferred `function ()`. What is the motivation for this change?

Comment (cmatheson) [reply index.js]: prettier doesn't support `function ()` (see https://github.com/prettier/prettier/issues/3161 and https://github.com/prettier/prettier/issues/3847)

Comment (lwintch) [reply index.js]: Keeping this request in mind.  Hopefully we can configure prettier in the future to match our linting rules, but for now the space after the `function` keyword is intentional.

**Comment by lwintch** [comment]
We are going to leave the linting rules as they are for now regarding this rule (space-before-function-paren) as the decision to have the space was intentional.

**Pull request closed without merging**